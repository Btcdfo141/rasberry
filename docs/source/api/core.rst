.. _core_module:

:mod:`spencerassistant.core`
-------------------------

.. automodule:: spencerassistant.core
    :members: