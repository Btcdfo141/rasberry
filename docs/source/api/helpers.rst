:mod:`spencerassistant.helpers`
============================

.. automodule:: spencerassistant.helpers
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.aiohttp\_client
-------------------------------------

.. automodule:: spencerassistant.helpers.aiohttp_client
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.area\_registry
------------------------------------

.. automodule:: spencerassistant.helpers.area_registry
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.check\_config
-----------------------------------

.. automodule:: spencerassistant.helpers.check_config
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.collection
--------------------------------

.. automodule:: spencerassistant.helpers.collection
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.condition
-------------------------------

.. automodule:: spencerassistant.helpers.condition
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.config\_entry\_flow
-----------------------------------------

.. automodule:: spencerassistant.helpers.config_entry_flow
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.config\_entry\_oauth2\_flow
-------------------------------------------------

.. automodule:: spencerassistant.helpers.config_entry_oauth2_flow
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.config\_validation
----------------------------------------

.. automodule:: spencerassistant.helpers.config_validation
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.data\_entry\_flow
---------------------------------------

.. automodule:: spencerassistant.helpers.data_entry_flow
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.debounce
------------------------------

.. automodule:: spencerassistant.helpers.debounce
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.deprecation
---------------------------------

.. automodule:: spencerassistant.helpers.deprecation
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.device\_registry
--------------------------------------

.. automodule:: spencerassistant.helpers.device_registry
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.discovery
-------------------------------

.. automodule:: spencerassistant.helpers.discovery
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.dispatcher
--------------------------------

.. automodule:: spencerassistant.helpers.dispatcher
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.entity
----------------------------

.. automodule:: spencerassistant.helpers.entity
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.entity\_component
---------------------------------------

.. automodule:: spencerassistant.helpers.entity_component
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.entity\_platform
--------------------------------------

.. automodule:: spencerassistant.helpers.entity_platform
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.entity\_registry
--------------------------------------

.. automodule:: spencerassistant.helpers.entity_registry
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.entity\_values
------------------------------------

.. automodule:: spencerassistant.helpers.entity_values
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.entityfilter
----------------------------------

.. automodule:: spencerassistant.helpers.entityfilter
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.event
---------------------------

.. automodule:: spencerassistant.helpers.event
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.icon
--------------------------

.. automodule:: spencerassistant.helpers.icon
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.integration\_platform
-------------------------------------------

.. automodule:: spencerassistant.helpers.integration_platform
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.intent
----------------------------

.. automodule:: spencerassistant.helpers.intent
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.json
--------------------------

.. automodule:: spencerassistant.helpers.json
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.location
------------------------------

.. automodule:: spencerassistant.helpers.location
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.logging
-----------------------------

.. automodule:: spencerassistant.helpers.logging
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.network
-----------------------------

.. automodule:: spencerassistant.helpers.network
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.restore\_state
------------------------------------

.. automodule:: spencerassistant.helpers.restore_state
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.script
----------------------------

.. automodule:: spencerassistant.helpers.script
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.service
-----------------------------

.. automodule:: spencerassistant.helpers.service
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.signal
-----------------------------

.. automodule:: spencerassistant.helpers.signal
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.state
---------------------------

.. automodule:: spencerassistant.helpers.state
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.storage
-----------------------------

.. automodule:: spencerassistant.helpers.storage
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.sun
-------------------------

.. automodule:: spencerassistant.helpers.sun
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.system\_info
----------------------------------

.. automodule:: spencerassistant.helpers.system_info
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.temperature
---------------------------------

.. automodule:: spencerassistant.helpers.temperature
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.template
------------------------------

.. automodule:: spencerassistant.helpers.template
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.translation
---------------------------------

.. automodule:: spencerassistant.helpers.translation
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.typing
----------------------------

.. automodule:: spencerassistant.helpers.typing
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.update\_coordinator
-----------------------------------------

.. automodule:: spencerassistant.helpers.update_coordinator
   :members:
   :undoc-members:
   :show-inheritance:
