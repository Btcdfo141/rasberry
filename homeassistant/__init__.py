"""Init file for Home Assistant."""


test2
