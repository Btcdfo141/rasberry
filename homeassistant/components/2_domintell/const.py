"""Constants for the Domintell Integration integration."""

DOMAIN = "2_domintell"
