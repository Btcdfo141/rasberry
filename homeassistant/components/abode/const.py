"""Constants for the Abode Security System component."""
import logging

LOGGER = logging.getLogger(__package__)

DOMAIN = "abode"
ATTRIBUTION = "Data provided by goabode.com"

DEFAULT_CACHEDB = "jaraco_abode_cache.pickle"
CONF_POLLING = "polling"
