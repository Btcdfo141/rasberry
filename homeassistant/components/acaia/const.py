"""Constants for component."""
from typing import Final

DOMAIN: Final = "acaia"
CONF_IS_NEW_STYLE_SCALE: Final = "is_new_style_scale"
