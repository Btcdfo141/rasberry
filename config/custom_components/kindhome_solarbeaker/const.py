from datetime import timedelta

from homeassistant.const import Platform

DOMAIN="kindhome_solarbeaker"
PLATFORMS = [Platform.COVER]

UPDATE_INTERVAL = timedelta(seconds=20)

SERVICE_UUID = "75c276c3-8f97-20bc-a143-b354244886d4"
TITLE="Kindhome Solarbeaker"

DATA_DEVICE = "device"
DATA_COOR = "coordinator"
