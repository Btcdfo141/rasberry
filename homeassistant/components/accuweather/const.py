"""Constants for AccuWeather integration."""
from __future__ import annotations

from typing import Final

from homeassistant.components.weather import (
    ATTR_CONDITION_CLEAR_NIGHT,
    ATTR_CONDITION_CLOUDY,
    ATTR_CONDITION_EXCEPTIONAL,
    ATTR_CONDITION_FOG,
    ATTR_CONDITION_HAIL,
    ATTR_CONDITION_LIGHTNING,
    ATTR_CONDITION_LIGHTNING_RAINY,
    ATTR_CONDITION_PARTLYCLOUDY,
    ATTR_CONDITION_POURING,
    ATTR_CONDITION_RAINY,
    ATTR_CONDITION_SNOWY,
    ATTR_CONDITION_SNOWY_RAINY,
    ATTR_CONDITION_SUNNY,
    ATTR_CONDITION_WINDY,
)
from homeassistant.const import (
    CONCENTRATION_PARTS_PER_CUBIC_METER,
    DEVICE_CLASS_TEMPERATURE,
    LENGTH_FEET,
    LENGTH_INCHES,
    LENGTH_METERS,
    LENGTH_MILLIMETERS,
    PERCENTAGE,
    SPEED_KILOMETERS_PER_HOUR,
    SPEED_MILES_PER_HOUR,
    TEMP_CELSIUS,
    TEMP_FAHRENHEIT,
    TIME_HOURS,
    UV_INDEX,
)

from .model import SensorDescription

ATTRIBUTION: Final = "Data provided by AccuWeather"
ATTR_FORECAST: Final = "forecast"
CONF_FORECAST: Final = "forecast"
COORDINATOR: Final = "coordinator"
DOMAIN: Final = "accuweather"
MANUFACTURER: Final = "AccuWeather, Inc."
NAME: Final = "AccuWeather"
UNDO_UPDATE_LISTENER: Final = "undo_update_listener"

CONDITION_CLASSES: Final[dict[str, list[int]]] = {
    ATTR_CONDITION_CLEAR_NIGHT: [33, 34, 37],
    ATTR_CONDITION_CLOUDY: [7, 8, 38],
    ATTR_CONDITION_EXCEPTIONAL: [24, 30, 31],
    ATTR_CONDITION_FOG: [11],
    ATTR_CONDITION_HAIL: [25],
    ATTR_CONDITION_LIGHTNING: [15],
    ATTR_CONDITION_LIGHTNING_RAINY: [16, 17, 41, 42],
    ATTR_CONDITION_PARTLYCLOUDY: [3, 4, 6, 35, 36],
    ATTR_CONDITION_POURING: [18],
    ATTR_CONDITION_RAINY: [12, 13, 14, 26, 39, 40],
    ATTR_CONDITION_SNOWY: [19, 20, 21, 22, 23, 43, 44],
    ATTR_CONDITION_SNOWY_RAINY: [29],
    ATTR_CONDITION_SUNNY: [1, 2, 5],
    ATTR_CONDITION_WINDY: [32],
}

FORECAST_DAYS: Final[list[int]] = [0, 1, 2, 3, 4]

FORECAST_SENSOR_TYPES: Final[dict[str, SensorDescription]] = {
    "CloudCoverDay": {
        "device_class": None,
        "icon": "mdi:weather-cloudy",
        "label": "Cloud Cover Day",
        "unit_metric": PERCENTAGE,
        "unit_imperial": PERCENTAGE,
    },
    "CloudCoverNight": {
        "device_class": None,
        "icon": "mdi:weather-cloudy",
        "label": "Cloud Cover Night",
        "unit_metric": PERCENTAGE,
        "unit_imperial": PERCENTAGE,
    },
    "Grass": {
        "device_class": None,
        "icon": "mdi:grass",
        "label": "Grass Pollen",
        "unit_metric": CONCENTRATION_PARTS_PER_CUBIC_METER,
        "unit_imperial": CONCENTRATION_PARTS_PER_CUBIC_METER,
    },
    "HoursOfSun": {
        "device_class": None,
        "icon": "mdi:weather-partly-cloudy",
        "label": "Hours Of Sun",
        "unit_metric": TIME_HOURS,
        "unit_imperial": TIME_HOURS,
    },
    "Mold": {
        "device_class": None,
        "icon": "mdi:blur",
        "label": "Mold Pollen",
        "unit_metric": CONCENTRATION_PARTS_PER_CUBIC_METER,
        "unit_imperial": CONCENTRATION_PARTS_PER_CUBIC_METER,
    },
    "Ozone": {
        "device_class": None,
        "icon": "mdi:vector-triangle",
        "label": "Ozone",
        "unit_metric": None,
        "unit_imperial": None,
    },
    "Ragweed": {
        "device_class": None,
        "icon": "mdi:sprout",
        "label": "Ragweed Pollen",
        "unit_metric": CONCENTRATION_PARTS_PER_CUBIC_METER,
        "unit_imperial": CONCENTRATION_PARTS_PER_CUBIC_METER,
    },
    "RealFeelTemperatureMax": {
        "device_class": DEVICE_CLASS_TEMPERATURE,
        "icon": None,
        "label": "RealFeel Temperature Max",
        "unit_metric": TEMP_CELSIUS,
        "unit_imperial": TEMP_FAHRENHEIT,
    },
    "RealFeelTemperatureMin": {
        "device_class": DEVICE_CLASS_TEMPERATURE,
        "icon": None,
        "label": "RealFeel Temperature Min",
        "unit_metric": TEMP_CELSIUS,
        "unit_imperial": TEMP_FAHRENHEIT,
    },
    "RealFeelTemperatureShadeMax": {
        "device_class": DEVICE_CLASS_TEMPERATURE,
        "icon": None,
        "label": "RealFeel Temperature Shade Max",
        "unit_metric": TEMP_CELSIUS,
        "unit_imperial": TEMP_FAHRENHEIT,
    },
    "RealFeelTemperatureShadeMin": {
        "device_class": DEVICE_CLASS_TEMPERATURE,
        "icon": None,
        "label": "RealFeel Temperature Shade Min",
        "unit_metric": TEMP_CELSIUS,
        "unit_imperial": TEMP_FAHRENHEIT,
    },
    "ThunderstormProbabilityDay": {
        "device_class": None,
        "icon": "mdi:weather-lightning",
        "label": "Thunderstorm Probability Day",
        "unit_metric": PERCENTAGE,
        "unit_imperial": PERCENTAGE,
    },
    "ThunderstormProbabilityNight": {
        "device_class": None,
        "icon": "mdi:weather-lightning",
        "label": "Thunderstorm Probability Night",
        "unit_metric": PERCENTAGE,
        "unit_imperial": PERCENTAGE,
    },
    "Tree": {
        "device_class": None,
        "icon": "mdi:tree-outline",
        "label": "Tree Pollen",
        "unit_metric": CONCENTRATION_PARTS_PER_CUBIC_METER,
        "unit_imperial": CONCENTRATION_PARTS_PER_CUBIC_METER,
    },
    "UVIndex": {
        "device_class": None,
        "icon": "mdi:weather-sunny",
        "label": "UV Index",
        "unit_metric": UV_INDEX,
        "unit_imperial": UV_INDEX,
    },
    "WindGustDay": {
        "device_class": None,
        "icon": "mdi:weather-windy",
        "label": "Wind Gust Day",
        "unit_metric": SPEED_KILOMETERS_PER_HOUR,
        "unit_imperial": SPEED_MILES_PER_HOUR,
    },
    "WindGustNight": {
        "device_class": None,
        "icon": "mdi:weather-windy",
        "label": "Wind Gust Night",
        "unit_metric": SPEED_KILOMETERS_PER_HOUR,
        "unit_imperial": SPEED_MILES_PER_HOUR,
    },
    "WindDay": {
        "device_class": None,
        "icon": "mdi:weather-windy",
        "label": "Wind Day",
        "unit_metric": SPEED_KILOMETERS_PER_HOUR,
        "unit_imperial": SPEED_MILES_PER_HOUR,
    },
    "WindNight": {
        "device_class": None,
        "icon": "mdi:weather-windy",
        "label": "Wind Night",
        "unit_metric": SPEED_KILOMETERS_PER_HOUR,
        "unit_imperial": SPEED_MILES_PER_HOUR,
    },
}

OPTIONAL_SENSORS: Final[tuple[str, ...]] = (
    "ApparentTemperature",
    "CloudCover",
    "CloudCoverDay",
    "CloudCoverNight",
    "DewPoint",
    "Grass",
    "Mold",
    "Ozone",
    "Ragweed",
    "RealFeelTemperatureShade",
    "RealFeelTemperatureShadeMax",
    "RealFeelTemperatureShadeMin",
    "Tree",
    "WetBulbTemperature",
    "WindChillTemperature",
    "WindGust",
    "WindGustDay",
    "WindGustNight",
)

SENSOR_TYPES: Final[dict[str, SensorDescription]] = {
    "ApparentTemperature": {
        "device_class": DEVICE_CLASS_TEMPERATURE,
        "icon": None,
        "label": "Apparent Temperature",
        "unit_metric": TEMP_CELSIUS,
        "unit_imperial": TEMP_FAHRENHEIT,
    },
    "Ceiling": {
        "device_class": None,
        "icon": "mdi:weather-fog",
        "label": "Cloud Ceiling",
        "unit_metric": LENGTH_METERS,
        "unit_imperial": LENGTH_FEET,
    },
    "CloudCover": {
        "device_class": None,
        "icon": "mdi:weather-cloudy",
        "label": "Cloud Cover",
        "unit_metric": PERCENTAGE,
        "unit_imperial": PERCENTAGE,
    },
    "DewPoint": {
        "device_class": DEVICE_CLASS_TEMPERATURE,
        "icon": None,
        "label": "Dew Point",
        "unit_metric": TEMP_CELSIUS,
        "unit_imperial": TEMP_FAHRENHEIT,
    },
    "RealFeelTemperature": {
        "device_class": DEVICE_CLASS_TEMPERATURE,
        "icon": None,
        "label": "RealFeel Temperature",
        "unit_metric": TEMP_CELSIUS,
        "unit_imperial": TEMP_FAHRENHEIT,
    },
    "RealFeelTemperatureShade": {
        "device_class": DEVICE_CLASS_TEMPERATURE,
        "icon": None,
        "label": "RealFeel Temperature Shade",
        "unit_metric": TEMP_CELSIUS,
        "unit_imperial": TEMP_FAHRENHEIT,
    },
    "Precipitation": {
        "device_class": None,
        "icon": "mdi:weather-rainy",
        "label": "Precipitation",
        "unit_metric": LENGTH_MILLIMETERS,
        "unit_imperial": LENGTH_INCHES,
    },
    "PressureTendency": {
        "device_class": "accuweather__pressure_tendency",
        "icon": "mdi:gauge",
        "label": "Pressure Tendency",
        "unit_metric": None,
        "unit_imperial": None,
    },
    "UVIndex": {
        "device_class": None,
        "icon": "mdi:weather-sunny",
        "label": "UV Index",
        "unit_metric": UV_INDEX,
        "unit_imperial": UV_INDEX,
    },
    "WetBulbTemperature": {
        "device_class": DEVICE_CLASS_TEMPERATURE,
        "icon": None,
        "label": "Wet Bulb Temperature",
        "unit_metric": TEMP_CELSIUS,
        "unit_imperial": TEMP_FAHRENHEIT,
    },
    "WindChillTemperature": {
        "device_class": DEVICE_CLASS_TEMPERATURE,
        "icon": None,
        "label": "Wind Chill Temperature",
        "unit_metric": TEMP_CELSIUS,
        "unit_imperial": TEMP_FAHRENHEIT,
    },
    "Wind": {
        "device_class": None,
        "icon": "mdi:weather-windy",
        "label": "Wind",
        "unit_metric": SPEED_KILOMETERS_PER_HOUR,
        "unit_imperial": SPEED_MILES_PER_HOUR,
    },
    "WindGust": {
        "device_class": None,
        "icon": "mdi:weather-windy",
        "label": "Wind Gust",
        "unit_metric": SPEED_KILOMETERS_PER_HOUR,
        "unit_imperial": SPEED_MILES_PER_HOUR,
    },
}
