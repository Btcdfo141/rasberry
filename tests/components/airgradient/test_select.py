"""Tests for the AirGradient select platform."""

from unittest.mock import AsyncMock, patch

import pytest
from syrupy import SnapshotAssertion

from homeassistant.const import Platform
from homeassistant.core import HomeAssistant
from homeassistant.helpers import entity_registry as er

from . import setup_integration

from tests.common import MockConfigEntry, snapshot_platform


@pytest.mark.usefixtures("entity_registry_enabled_by_default")
async def test_all_entities(
    hass: HomeAssistant,
    snapshot: SnapshotAssertion,
    mock_airgradient_client: AsyncMock,
    mock_config_entry: MockConfigEntry,
    entity_registry: er.EntityRegistry,
) -> None:
    """Test all entities."""
    with patch("homeassistant.components.airgradient.PLATFORMS", [Platform.SELECT]):
        await setup_integration(hass, mock_config_entry)

    await snapshot_platform(hass, entity_registry, snapshot, mock_config_entry.entry_id)
