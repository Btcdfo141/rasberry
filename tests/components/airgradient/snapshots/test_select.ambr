# serializer version: 1
# name: test_all_entities[select.airgradient_configuration_source-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'options': list([
        'cloud',
        'local',
        'both',
      ]),
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'select',
    'entity_category': None,
    'entity_id': 'select.airgradient_configuration_source',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Configuration source',
    'platform': 'airgradient',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'configuration_control',
    'unique_id': '84fce612f5b8-configuration_control',
    'unit_of_measurement': None,
  })
# ---
# name: test_all_entities[select.airgradient_configuration_source-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Airgradient Configuration source',
      'options': list([
        'cloud',
        'local',
        'both',
      ]),
    }),
    'context': <ANY>,
    'entity_id': 'select.airgradient_configuration_source',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unknown',
  })
# ---
# name: test_all_entities[select.airgradient_none-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'options': list([
        'c',
        'f',
      ]),
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'select',
    'entity_category': None,
    'entity_id': 'select.airgradient_none',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': None,
    'platform': 'airgradient',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'temperature_unit',
    'unique_id': '84fce612f5b8-temperature_unit',
    'unit_of_measurement': None,
  })
# ---
# name: test_all_entities[select.airgradient_none-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Airgradient None',
      'options': list([
        'c',
        'f',
      ]),
    }),
    'context': <ANY>,
    'entity_id': 'select.airgradient_none',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unknown',
  })
# ---
